### ISSUE TITLE
circular argument references in adopter_searcher

### ISSUE THREAD

**Issue by markottaviani** [ophrescue/RescueRails#1637]
Warnings throwing during rspec run

```
app/models/adopter_searcher.rb:32: warning: circular argument reference - user_id
app/models/adopter_searcher.rb:68: warning: circular argument reference - user_id
```

Looks like this was introduced with  64a217f29d1dec28edd8ed5657898f819c2844b2

**Comment by markottaviani** [comment]
@danisaurio do you want to check this out?

**Comment by danisaurio** [comment]
I'll do it, sorry for the delay, I was busy. I'm on it

**Comment by markottaviani** [comment]
@danisaurio had to revert the circular argument fix  due to a bug #1662 , but it did find a hole in our test suite!

I've added a new test case to `adopter_searcher_spec`  that should verify the next attempt.
